qiskit_nature/algorithms/ground_state_solvers/minimum_eigensolver_factories/vqe_ucc_factory.py

```diff
@@ -81,108 +86,159 @@ def __init__(
                 ansatz, the evaluated mean and the evaluated standard deviation.`
             kwargs: any additional keyword arguments will be passed on to the VQE.
         """
-        self.quantum_instance = quantum_instance
-        self.optimizer = optimizer
-        self.initial_point = initial_point
-        self.gradient = gradient
-        self.expectation = expectation
-        self.include_custom = include_custom
-        self.ansatz = ansatz
-        self.initial_state = initial_state
-        self.callback = callback
-        self._kwargs = kwargs
+        self._ansatz = ansatz
+        self._initial_state = initial_state
 
-    @property
+        self._vqe = VQE(
+            quantum_instance=quantum_instance,
+            optimizer=optimizer,
+            initial_point=initial_point,
+            gradient=gradient,
+            expectation=expectation,
+            include_custom=include_custom,
+            callback=callback,
+            **kwargs,
```

Comment:
Since this factory is made specifically to handle the `ansatz` and `initial_point`, I believe it would be good if we deprecate all other explicit keyword arguments of this initializer in favor of `**kwargs`. That way we do not need to keep this method as much in sync with the underlying `VQE`.

The ansatz and initial_state should still remain configurable via this initializer, but should immediately set the `self._vqe` attributes.

Reply:
So, I believe that in order to implement this we need to loose the positional arguments that can use in the current implementations. Code like VQEUCCFactory(some_quantum_instance) wouldn't work anymore and should be replaced by VQEUCCFactory(quantum_instance = some_quantum_instance). Otherwise the constructor will think of some_quantum_instance as an ansatz.
 It is a small sacrifice, but it would mean that some people would need to change their code.

Reply:
Since all arguments of the `VQEUCCFactory` are keyword arguments, we do not give any guarantee for code to work when used via positional arguments. As such this should not be a problem. That we internally use the factories via positional arguments internally (e.g. during unittests) needs to be updated, yes

Reply:
Perfect, I have already changed the unittests and they work fine.